test/adapter/json_api/parse_test.rb

```diff
@@ -130,6 +130,25 @@ def test_polymorphic
             }
             assert_equal(expected, parsed_hash)
           end
+
+          def test_sti_type
+            parsed_hash = ActiveModelSerializers::Adapter::JsonApi::Deserialization.parse!(@hash, sti_type_key: :_type)
+            assert_equal(@expected.merge(_type: 'Photo'), parsed_hash)
+
+            with_config(sti_type_key: :_type) do
+              parsed_hash = ActiveModelSerializers::Adapter::JsonApi::Deserialization.parse!(@hash)
+              assert_equal(@expected.merge(_type: 'Photo'), parsed_hash)
+            end
+          end
+
+          def test_namespaced_sti_type
+            hash = @hash
+            hash['data']['type'] = 'content--user-photos'
```

Comment:
Thanks for the PR.  So, if I can understand the issue:

Given the resource

```ruby
{
  id: '1',
  type: 'content--user-photos',
  attributes: { title: 'A day at the beach' }
}
```

When we deserialize it we get

```ruby
{
  id: 1,
  title: 'A day at the beach' 
}
```

and lose the resource type in the resulting attributes.

In Rails, there is an single-table-inheritance (sti) attribute used to determine the class to instantiate from the database. The default name for this field is `type`, which is an `attribute` on the ActiveRecord object.

When AMS serializes a User it uses a UserSerializer with the default JSON API resource type of `users`.  Similarly, the default behavior would be UserPhoto to use the UserPhotoSerializer and have the JSON API resource type of `user-photos`. In both cases, unless specified in the serializer, the `type` field would not be serialized as an attribute.

So, I'm thinking that part of the use case here is what you do with the deserialized JSON, and if you can or should use the resource type vs. the attribute 'type'. Is that right?  

How are you using the deserialized JSON document to interact with ActiveRecord?